src/control/lib/hardware/topology.go

```diff
@@ -353,8 +353,8 @@ func (t *Topology) Merge(newTopo *Topology) error {
 							oldDev.Type = newDev.Type
 						}
 
-						if oldDev.LinkSpeed == 0 {
-							oldDev.LinkSpeed = newDev.LinkSpeed
+						if oldDev.LinkNegSpeed == 0 {
+							oldDev.LinkNegSpeed = newDev.LinkNegSpeed
 						}
```

Comment:
Just a reminder that you'll need to handle the new fields here, too.

Reply:
done